polkadot/runtime/parachains/src/inclusion/mod.rs

```diff
@@ -935,6 +941,23 @@ impl<T: Config> Pallet<T> {
 		para: ParaId,
 		upward_messages: &[UpwardMessage],
 	) -> Result<(), UmpAcceptanceCheckErr> {
+		// Filter any pending UMP signals and the separator.
+		let upward_messages = if let Some(separator_index) =
+			upward_messages.iter().rposition(|message| message.is_empty())
+		{
+			let (upward_messages, ump_signals) = upward_messages.split_at(separator_index);
+
+			if ump_signals.len() > 2 {
```

Comment:
is 0 ump_signals allowed in case there's a separator?
is multiple separators allowed? we only look here for the last one
my thinking it should be no in both cases - might be good to validate those

Reply:
0 is not allowed and this should be tested by the `check_core_index` function and `selected_core` should return `None`. I will have to double check and add another check here if this is not covered.

If we have multiple separators that is fine, it means that the parachain wants to hurt itself. Those will count as XCM messages and are limited by a param in relay chain configuration.

Reply:
Currently it seems we are not really strict, we just consider that the candidate does not commit to a core as we are using Option.  I will add stricter checks.

Reply:
Made check stricter and added test for separator + 0 ump signals .